optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -526,36 +526,380 @@ private static Runnable innerCountDistinctLong(CountDistinctLongResultContainer
     // min
     // ************************************************************************
 
+    /**
+     * Returns a collector that finds a minimum value in a group of {@link Comparable} elements.
+     * <p>
+     *     It is expected that the {@link Comparable}'s {@link Comparable#compareTo(Object)} is consistent with
+     *     {@link #equals(Object)}.
+     *     In other words, if two elements compare to equal, any of them may be returned by the collector as they are
+     *     considered to be {@link #equals(Object)}.
+     * <p>
+     *     Example: Assume the following elements of type Person: Ann(age = 20), Beth(age = 25), Cathy(age = 30),
+     *     David(age = 25), Eric(age = 20).
+     *     Further assume that the Person type compares by age.
+     *     Such collector returns either Ann or Eric as the minimum element.
```

Comment:
Use `{@code Ann(age = 20)}` to format example input and output.
Wrap a collection of elements as input or output in `[]`.
Use formal (pseudo-)code-snippet `.groupBy(min(Person::getAge))` instead of assume that the Person type compares by age.
Find a shorter template for "input + constraintstream = output". The "Such collector" breaks this flow (too much ceremony).

Reply:
Proposal:

```
For example, given `[Ann(age = 20), Beth(age = 25), Cathy(age = 30), David(age = 25), Eric(age = 20)]` with `.groupBy(min(Comparator.comparing(Person::getAge)))`, this returns `Ann` or `Eric` arbitrary.
```

Reply:
If this template proposal works well, let's apply it to all of constraint streams for all example:

"For example, {@code <input>} with {@code <cs snippet>} returns {@code <output>}."

Reply:
Also,
in ConstraintStream.join(Class) we have a hint that they probably want to use join(Class, Joiners) instead if they can. Let's do something similar here, that links to min(Function).

Reply:
And before the hint, let's recommend the good practice of ending with an identity comparison, by adding a 2th example with `.groupBy(min(Comparator.comparing(Person::getAge).comparing(Person::getId)))`